### PR TITLE
fix the issue with crunchy operator for trial version cluster creation

### PR DESCRIPTION
**Pull request by redhatHameed**
Signed-off-by: Abdul Hameed <ahameed@redhat.com>

setting the default value for cloud provider region parameter `us-east-1` for crunchy bridge operator trial version, as the dbaas instance cr only set cloud provider default aws. 

```spec:
  inventoryRef:
    name: ahameed-r5
    namespace: openshift-dbaas-operator
  provisioningParameters:
    cloudProvider: aws
    name: dbaastest
    plan: FREETRIAL
    teamProject: '': ''
```

setting region and removing the teamProject from cr, as it's not required for trail version, and it will get default.   
```
spec:
  cloudProvider: aws
  cloudRegion: us-east-1
  inventoryRef:
    name: ahameed-r5
    namespace: openshift-dbaas-operator
  name: dbaastest
  otherInstanceParams:
    Name: dbaastest
    Provider: aws
```

Removing the TeamID parameters as these are not required and avoid passing empty or any values from UI to crunchy operator, to work trial version as release4. 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA or in issue number have been completed
- [ ] Unit tests added that prove the fix is effective or the feature works 
- [ ] Documentation added for the feature
- [ ] all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

File: api/v1alpha1/dbaasinstance_conversion.go [hunk 1]
```diff
@@ -99,8 +99,8 @@ func (src *DBaaSInstanceSpec) ConvertTo(dst *v1beta1.DBaaSInstanceSpec) error {
 // ConvertFrom converts the DBaaSInstanceSpec from the v1beta1 to this version.
 func (dst *DBaaSInstanceSpec) ConvertFrom(src *v1beta1.DBaaSInstanceSpec) error {
 	dst.Name = src.ProvisioningParameters[v1beta1.ProvisioningName]
-	dst.CloudProvider = src.ProvisioningParameters[v1beta1.ProvisioningCloudProvider]
-	dst.CloudRegion = src.ProvisioningParameters[v1beta1.ProvisioningRegions]
+	//dst.CloudProvider = src.ProvisioningParameters[v1beta1.ProvisioningCloudProvider]
+	//dst.CloudRegion = src.ProvisioningParameters[v1beta1.ProvisioningRegions]
```
Comment (tchughesiv) [inline api/v1alpha1/dbaasinstance_conversion.go:103]: would a better approach be to only set if not empty?

File: api/v1alpha1/dbaasinstance_conversion.go [hunk 2]
```diff
@@ -99,8 +99,8 @@ func (src *DBaaSInstanceSpec) ConvertTo(dst *v1beta1.DBaaSInstanceSpec) error {
 // ConvertFrom converts the DBaaSInstanceSpec from the v1beta1 to this version.
 func (dst *DBaaSInstanceSpec) ConvertFrom(src *v1beta1.DBaaSInstanceSpec) error {
 	dst.Name = src.ProvisioningParameters[v1beta1.ProvisioningName]
-	dst.CloudProvider = src.ProvisioningParameters[v1beta1.ProvisioningCloudProvider]
-	dst.CloudRegion = src.ProvisioningParameters[v1beta1.ProvisioningRegions]
+	//dst.CloudProvider = src.ProvisioningParameters[v1beta1.ProvisioningCloudProvider]
```
Comment (jianrongzhang89) [inline api/v1alpha1/dbaasinstance_conversion.go:102]: Why this is removed? Only TeamID needs to be removed, right?

Comment (redhatHameed) [reply api/v1alpha1/dbaasinstance_conversion.go]: updated, let me know WDYT ?

File: api/v1alpha1/dbaasinstance_conversion.go (hunk 1 above)
Comment (redhatHameed) [reply api/v1alpha1/dbaasinstance_conversion.go]: updated, let me know WDYT ?

**Review by jianrongzhang89 — APPROVED** [review]
LGTM

**Pull request merged**